reinforcement_learning/common/sagemaker_rl/ray_launcher.py

```diff
@@ -231,6 +240,47 @@ def save_checkpoint_and_serving_model(self, algorithm=None, env_string=None, con
         # To ensure SageMaker local mode works fine
         change_permissions_recursive(INTERMEDIATE_DIR, 0o777)
         change_permissions_recursive(MODEL_OUTPUT_DIR, 0o777)
+        
+    def set_up_checkpoint(self, config=None):
+        try:
+            checkpoint_dir = config['training']['restore']
+            print("Found checkpoint dir in user config.")
+        except KeyError:
```

Comment:
Same here, you can `.get` to avoid `KeyError` Also not quite understand this line, it seems like no matter what happened here, `checkpoint_dir ` will be reset into "" in line 257

Reply:
Customer can restore the experiment in two ways: directly specifying the `restore` config in training script, or putting the checkpoint files in CHECKPOINT channel and let SageMaker take care of it. There is an edge case where customer provides both and I'm testing that.